Orange/widgets/visualize/owscatterplotgraph.py

```diff
@@ -766,7 +767,8 @@ def __call__(self):
                     widget.scatterplot_item.setSize(size)
                     widget.scatterplot_item_sel.setSize(size + SELECTION_WIDTH)
 
-            if np.sum(current_size_data) / self.n_valid != -1 and np.sum(diff):
+            if np.sum(current_size_data) / self.n_valid != -1 and \
+                    max(abs(diff)) and self.n_valid <= MAX_N_VALID_SIZE_ANIMATE:
```

Comment:
Can we make this condition clearer?
a) `max(abs(diff))` if better then before, but `np.any(diff != 0)` would be even better / more explicit
b) As far as I can see, sizes are always positive. So the first check could just be `np.all(current_size_data > 0)` i.e. legit sizes, or you can check for default initialization directly `np.any(current_size_data == -1)`
c) I would put the last check `<= MAX_N...` in first place so that for large data other checks don't need to be performed at all.